src/python_testing/TC_ACL_2_11.py

```diff
@@ -0,0 +1,172 @@
+# === END CI TEST ARGUMENTS ===
+
+import logging
+import queue
+
+import chip.clusters as Clusters
+from chip.clusters.Attribute import EventReadResult, SubscriptionTransaction
+from chip.clusters.ClusterObjects import ALL_ACCEPTED_COMMANDS, ALL_ATTRIBUTES, ALL_CLUSTERS, ClusterEvent
+from chip.clusters.Objects import AccessControl
+from chip.interaction_model import Status
+from matter_testing_support import MatterBaseTest, TestStep, async_test_body, default_matter_test_main
+from mobly import asserts
+
+
+class EventChangeCallback:
+    def __init__(self, expected_event: ClusterEvent, output: queue.Queue):
+        self._output = output
+        self._expected_cluster_id = expected_event.cluster_id
+        self._expected_event_id = expected_event.event_id
+
+    def __call__(self, res: EventReadResult, transaction: SubscriptionTransaction):
+        if res.Status == Status.Success and res.Header.ClusterId == self._expected_cluster_id and res.Header.EventId == self._expected_event_id:
+            logging.info(
+                f'Got subscription report for event {self._expected_event_id} on cluster {self._expected_cluster_id}: {res.Data}')
+            self._output.put(res)
+
+
+def WaitForEventReport(q: queue.Queue, expected_event: ClusterEvent):
+    try:
+        res = q.get(block=True, timeout=10)
+    except queue.Empty:
+        asserts.fail("Failed to receive a report for the event {}".format(expected_event))
+
+    asserts.assert_equal(res.Header.ClusterId, expected_event.cluster_id, "Expected cluster ID not found in event report")
+    asserts.assert_equal(res.Header.EventId, expected_event.event_id, "Expected event ID not found in event report")
+
+
+class TC_ACL_2_11(MatterBaseTest):
+
+    def desc_TC_ACL_2_11(self) -> str:
+        return "[TC-ACL-2.11] Verification of Managed Device feature"
+
+    def steps_TC_ACL_2_11(self) -> list[TestStep]:
+        steps = [
+            TestStep(1, "Commissioning, already done"),
+            TestStep(2, "TH1 reads DUT Endpoint 0 AccessControl cluster CommissioningARL attribute"),
+            TestStep(3, "TH1 reads DUT Endpoint 0 AccessControl cluster ARL attribute"),
+            TestStep(4, "For each entry in ARL, iterate over each restriction and attempt access the restriction's ID on the Endpoint and Cluster in the ARL entry.",
+                     "If the restriction is Type AttributeAccessForbidden, read the restriction's attribute ID and verify the response is UNSUPPORTED_ACCESS."
+                     "If the restriction is Type AttributeWriteForbidden, write restriction's the attribute ID and verify the response is UNSUPPORTED_ACCESS."
+                     "If the restriction is Type CommandForbidden, invoke the restriction's command ID and verify the response is UNSUPPORTED_ACCESS."),
```

Comment:
This is missing a negative test case that entries that should not be restricted are not restricted.

Reply:
Will fix in https://github.com/CHIP-Specifications/chip-test-plans/issues/4560